### ISSUE TITLE
Range indicator is incorrect.

### ISSUE THREAD

**Issue by rgswords** [SabineWren/Quiver#21]
![image](https://github.com/user-attachments/assets/6ebdf7c6-f7fc-4bac-8986-23d71824793f)

As you can see, this is NOT melee range. Incorrect value whether locked or unlocked, for all targets.

**Comment by SabineWren** [comment]
It works for my level 1 test character. How do you reproduce that?

![Screenshot from 2024-09-07 23-22-52](https://github.com/user-attachments/assets/0a50f154-ee65-49c4-93c6-7484b5e5bf2e)

**Comment by SabineWren** [comment]
Do you have ranged macros with the wing clip texture?

**Comment by rgswords** [comment]
I do have a macro with the Wing Clip texture, but it's melee abilities: Wing Clip and Raptor Strike.

**Comment by SabineWren** [comment]
Does it still happen without that macro on your bars? What about on a level 1 hunter?

**Comment by rgswords** [comment]
Taking the macro off the main bar apparently fixes the behavior in showing the correct ranges as I move towards and away from the target. Aside from that, there are no melee abilities/icons on any of my bars. The macro itself uses the Wing Clip icon and is a simple spammable:
_/cast Raptor Strike
/cast Wing Clip_

- Changing its icon to any other, including Raptor Strike, does allow the macro to be on the bars without affecting the range indicator.
- A test macro with the Wing Clip icon to _/say Testing icon_ does not break the range indicator.
- A test macro with the Wing Clip icon that casts Raptor Strike AFTER Wing Clip (i.e. reverse the lines of the original) does not break the range indicator.
- A test SuperMacro with the Wing Clip icon, first line #showtooltip Wing Clip and then the lines of the original macro also does not break the range indicator. (Not really usable as #showtooltip doesn't work quite correctly, having the character /say it)

Unfortunately I do not have room for a level 1 on this account. Please let me know if you have additional test cases.

**Comment by SabineWren** [comment]
Interesting... despite the action bar icon visibly changing based on macro contents, it seems `GetActionTexture(slot)` returns the initial texture set in the macro menu. The range check requires finding the spell on an action bar slot by texture. I'm able to create false positives by setting the initial texture to Wing Clip (Rogue Trip), then setting the macro to `/cast Aimed Shot`.

**Comment by SabineWren** [comment]
This isn't intuitive behavior. I could change it back to only looking at raw spells instead of macros.

**Comment by SabineWren** [comment]
Still confused though, since I can't replicate any of the points you mentioned. Regardless of raptor strike vs wing clip, the macro either has range-checked texture texture + any spell in range, or it doesn't. Regardless, requiring raw spellbook abilities on the action bar is much more consistent.

**Comment by SabineWren** [comment]
If you find the line:
```lua
local isSpell = ActionHasRange(i) or GetActionText(i) == nil
```
and change `or` to `and`. That should fix it. I'm not going to bump the version number over this, since it's likely a rare edge case.

**Comment by SabineWren** [comment]
Fixed on release branch e281abda6fce4db282683aa7f07b0781ab9c3bd4
